frontend/app/services/api-notifications-service.js

```diff
@@ -26,37 +26,26 @@
 // See doc/COPYRIGHT.rdoc for more details.
 //++
 
-// TODO move to UI components
-module.exports = function($rootScope, $timeout, ConfigurationService) {
+module.exports = function(NotificationsService, ApiHelper) {
+  'use strict';
 
-  return {
-    restrict: 'E',
-    replace: true,
-    scope: {},
-    templateUrl: '/templates/components/flash_message.html',
-    link: function(scope, element, attrs) {
-      $rootScope.$on('flashMessage', function(event, message) {
-        scope.message = message;
-        scope.flashType = 'notice';
-        scope.flashId = 'flash-notice';
+  var addError = function(error) {
+    var messages = ApiHelper.getErrorMessages(error);
 
-        var fadeOutTime = attrs.fadeOutTime || 3000;
+    if (messages.length > 1) {
+      NotificationsService.addError('', messages);
+    }
+    else {
+      NotificationsService.addError(messages[0], []);
```

Comment:
This if-else looks a little clumsy.
I think to remember how it was intended: "we have a list of error messages and maybe even an additional message", but this is not how you are using it at the moment.

Two (and a half) ideas:

* always use the errors list
    * either always passing a message (`addError('The following fancy errors occured', messages)`)
    * only passing the errors (second param) and deciding inside the `NotificationService` how to display the error message depending on error count and presence of a message
* consequently change the signature (and assumptions) of the `NotificationService` to support how we are now using it (e.g. it explicitly checks whether errors were passed, because it did **not** intend to have no errors)

My favorite option at the moment is option 1, since it seems more consistent to me...